core/os/os.h

```diff
@@ -535,6 +536,9 @@ class OS {
 	virtual bool request_permissions() { return true; }
 	virtual Vector<String> get_granted_permissions() const { return Vector<String>(); }
 
+	bool get_hide_home_indicator() const { return _hide_home_indicator; }
+	void set_hide_home_indicator(bool p_hide) { _hide_home_indicator = p_hide; }
```

Comment:
If you already have getter and setter, it should be exposed to gdscript as property in `core_bind`.

`set_hide_home_indicator` should be virtual, and it's iOS implementation call `[view_controller setNeedsUpdateOfHomeIndicatorAutoHidden];` to update home indicator state if this property is changed it in runtime.

```diff
diff --git a/core/bind/core_bind.cpp b/core/bind/core_bind.cpp
index 4c8dcc20ea..efac68631f 100644
--- a/core/bind/core_bind.cpp
+++ b/core/bind/core_bind.cpp
@@ -1165,6 +1165,16 @@ Vector<String> _OS::get_granted_permissions() const {
 	return OS::get_singleton()->get_granted_permissions();
 }
 
+bool _OS::get_hide_home_indicator() const {
+
+	return OS::get_singleton()->get_hide_home_indicator();
+}
+
+void _OS::set_hide_home_indicator(bool p_hide) {
+
+	OS::get_singleton()->set_hide_home_indicator(p_hide);
+}
+
 _OS *_OS::singleton = NULL;
 
 void _OS::_bind_methods() {
@@ -1357,6 +1367,9 @@ void _OS::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("request_permissions"), &_OS::request_permissions);
 	ClassDB::bind_method(D_METHOD("get_granted_permissions"), &_OS::get_granted_permissions);
 
+	ClassDB::bind_method(D_METHOD("get_hide_home_indicator"), &_OS::get_hide_home_indicator);
+	ClassDB::bind_method(D_METHOD("set_hide_home_indicator", "hide"), &_OS::set_hide_home_indicator);
+
 	ADD_PROPERTY(PropertyInfo(Variant::STRING, "clipboard"), "set_clipboard", "get_clipboard");
 	ADD_PROPERTY(PropertyInfo(Variant::INT, "current_screen"), "set_current_screen", "get_current_screen");
 	ADD_PROPERTY(PropertyInfo(Variant::INT, "exit_code"), "set_exit_code", "get_exit_code");
@@ -1378,6 +1391,8 @@ void _OS::_bind_methods() {
 	ADD_PROPERTY(PropertyInfo(Variant::VECTOR2, "window_position"), "set_window_position", "get_window_position");
 	ADD_PROPERTY(PropertyInfo(Variant::VECTOR2, "window_size"), "set_window_size", "get_window_size");
 
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "hide_home_indicator"), "set_hide_home_indicator", "get_hide_home_indicator");
+
 	// Those default values need to be specified for the docs generator,
 	// to avoid using values from the documentation writer's own OS instance.
 	ADD_PROPERTY_DEFAULT("clipboard", "");
diff --git a/core/bind/core_bind.h b/core/bind/core_bind.h
index d57da36ca5..d43be7c967 100644
--- a/core/bind/core_bind.h
+++ b/core/bind/core_bind.h
@@ -358,6 +358,9 @@ public:
 	bool request_permissions();
 	Vector<String> get_granted_permissions() const;
 
+	bool get_hide_home_indicator() const;
+	void set_hide_home_indicator(bool p_hide);
+
 	static _OS *get_singleton() { return singleton; }
 
 	_OS();
diff --git a/core/os/os.h b/core/os/os.h
index 26db629d7c..23a13e3dee 100644
--- a/core/os/os.h
+++ b/core/os/os.h
@@ -535,6 +535,9 @@ public:
 	virtual bool request_permissions() { return true; }
 	virtual Vector<String> get_granted_permissions() const { return Vector<String>(); }
 
-	bool get_hide_home_indicator() const { return _hide_home_indicator; }
-	void set_hide_home_indicator(bool p_hide) { _hide_home_indicator = p_hide; }
+	virtual bool get_hide_home_indicator() const { return _hide_home_indicator; }
+	virtual void set_hide_home_indicator(bool p_hide) { _hide_home_indicator = p_hide; }

 	virtual void process_and_drop_events() {}
 	OS();
 	virtual ~OS();
diff --git a/platform/iphone/app_delegate.mm b/platform/iphone/app_delegate.mm
index 3f1230faa8..d02529bbb4 100644
--- a/platform/iphone/app_delegate.mm
+++ b/platform/iphone/app_delegate.mm
@@ -66,6 +66,11 @@ void _vibrate() {
 	AudioServicesPlaySystemSound(kSystemSoundID_Vibrate);
 };
 
+void _update_home_indicator_auto_hidden() {
+	ViewController *root_controller = (ViewController *)((AppDelegate *)[[UIApplication sharedApplication] delegate]).window.rootViewController;
+	[root_controller setNeedsUpdateOfHomeIndicatorAutoHidden];
+};
+
 @implementation AppDelegate
 
 @synthesize window;
diff --git a/platform/iphone/os_iphone.cpp b/platform/iphone/os_iphone.cpp
index 8984ed1d7b..55fb8aa25b 100644
--- a/platform/iphone/os_iphone.cpp
+++ b/platform/iphone/os_iphone.cpp
@@ -538,6 +538,12 @@ extern void _pause_video();
 extern void _unpause_video();
 extern void _stop_video();
 extern void _focus_out_video();
+extern void _update_home_indicator_auto_hidden();
+
+void OSIPhone::set_hide_home_indicator(bool p_hide) {
+	OS::set_hide_home_indicator(p_hide);
+	_update_home_indicator_auto_hidden();
+}
 
 Error OSIPhone::native_video_play(String p_path, float p_volume, String p_audio_track, String p_subtitle_track) {
 	FileAccess *f = FileAccess::open(p_path, FileAccess::READ);
diff --git a/platform/iphone/os_iphone.h b/platform/iphone/os_iphone.h
index 1f49062cfc..477869a5fc 100644
--- a/platform/iphone/os_iphone.h
+++ b/platform/iphone/os_iphone.h
@@ -143,6 +143,8 @@ public:
 
 	static OSIPhone *get_singleton();
 
+	void set_hide_home_indicator(bool p_hide);
+
 	virtual void set_mouse_show(bool p_show);
 	virtual void set_mouse_grab(bool p_grab);
 	virtual bool is_mouse_grab_enabled() const;
```

Reply:
Is there an alternative way to implement this without adding yet another platform-specific method to `OS`? It really gets bloated with methods that most users don't need.

Reply:
Can't we just query the project setting in the iOS code and do things accordingly? It doesn't need to be changeable from scripts at runtime, as this likely requires an app restart anyway.

Reply:
If we don't want it to be changeable at all we can go this way. 
Does `GLOBAL_DEF` works in platform specific code, and does it appears in the editor then?

Reply:
> as this likely requires an app restart anyway.

Nop, it's changeable, I have tested it.

> Can't we just query the project setting in the iOS code and do things accordingly?

Reduce it to (+docs):
```diff

diff --git a/main/main.cpp b/main/main.cpp
index cdaee06135..3ac343a3c5 100644
--- a/main/main.cpp
+++ b/main/main.cpp
@@ -1126,6 +1126,8 @@ Error Main::setup(const char *execpath, int argc, char *argv[], bool p_second_ph
 	OS::get_singleton()->set_low_processor_usage_mode_sleep_usec(GLOBAL_DEF("application/run/low_processor_mode_sleep_usec", 6900)); // Roughly 144 FPS
 	ProjectSettings::get_singleton()->set_custom_property_info("application/run/low_processor_mode_sleep_usec", PropertyInfo(Variant::INT, "application/run/low_processor_mode_sleep_usec", PROPERTY_HINT_RANGE, "0,33200,1,or_greater")); // No negative numbers
 
+	GLOBAL_DEF("display/window/ios/hide_home_indicator", true);
+
 	Engine::get_singleton()->set_frame_delay(frame_delay);
 
 	message_queue = memnew(MessageQueue);
diff --git a/platform/iphone/view_controller.h b/platform/iphone/view_controller.h
index 68e3bc64fc..c2ef95acb5 100644
--- a/platform/iphone/view_controller.h
+++ b/platform/iphone/view_controller.h
@@ -45,4 +45,6 @@
 
 - (BOOL)prefersStatusBarHidden;
 
+- (BOOL)prefersHomeIndicatorAutoHidden;
+
 @end
diff --git a/platform/iphone/view_controller.mm b/platform/iphone/view_controller.mm
index e52ad92bf2..5bdce6277b 100644
--- a/platform/iphone/view_controller.mm
+++ b/platform/iphone/view_controller.mm
@@ -32,6 +32,8 @@
 
 #include "os_iphone.h"
 
+#include "core/project_settings.h"
+
 extern "C" {
 
 int add_path(int, char **);
@@ -129,6 +131,14 @@ int add_cmdline(int p_argc, char **p_args) {
 	return YES;
 }
 
+- (BOOL)prefersHomeIndicatorAutoHidden {
+	if (GLOBAL_GET("display/window/ios/hide_home_indicator")) {
+		return YES;
+	} else {
+		return NO;
+	}
+}
+
 #ifdef GAME_CENTER_ENABLED
 - (void)gameCenterViewControllerDidFinish:(GKGameCenterViewController *)gameCenterViewController {
 	//[gameCenterViewController dismissViewControllerAnimated:YES completion:^{GameCenter::get_singleton()->game_center_closed();}];//version for signaling when overlay is completely gone

}

```

Reply:
I like the last suggestion. IMO that's sufficient for this feature. If some users really want to change this at runtime for their app, we can consider adding methods to `OS` or hopefully a cleaner `iOS` singleton in the future, but for now I think it's better to wait for use cases.

Reply:
I agree, I have updated the PR with your suggestions. Thanks!